mir_eval/transcription.py

```diff
@@ -0,0 +1,406 @@
+        Array of estimated pitch values in Hertz
+    onset_tolerance : float > 0
+        The tolerance for an estimated note's onset deviating from the
+        reference note's onset, in seconds. Default is 0.05 (50 ms).
+    pitch_tolerance: float > 0
+        The tolerance for an estimated note's pitch deviating from the
+        reference note's pitch, in cents. Default is 50.0 (50 cents).
+    offset_ratio: float > 0 or None
+        The ratio of the reference note's duration used to define the
+        offset_tolerance. Default is 0.2 (20%), meaning the offset_tolerance
+        will equal the ref_duration * 0.2, or 0.05 (50 ms), whichever is
+        greater. If ``offset_ratio`` is set to ``None``, offsets are ignored in
+        the matching.
+    offset_min_tolerance: float > 0
+        The minimum tolerance for offset matching. See offset_ratio description
+        for an explanation of how the offset tolerance is determined. Note:
+        this parameter only influences the results if ``offset_ratio`` is not
+        ``None``.
+    strict: bool
+        If ``strict=False`` (the default), threshold checks for onset, offset,
+        and pitch matching are performed using ``<=`` (less than or equal). If
+        ``strict=True``, the threshold checks are performed using ``<`` (less
+        than).
+
+    Returns
+    -------
+    matching : list of tuples
+        A list of matched reference and estimated notes.
+        ``matching[i] == (i, j)`` where reference note i matches estimate note
+        j.
+    """
+    # set the comparison function
+    if strict:
+        cmp_func = np.less
+    else:
+        cmp_func = np.less_equal
+
+    # check for onset matches
+    onset_distances = np.abs(np.subtract.outer(ref_intervals[:, 0],
+                                               est_intervals[:, 0]))
+    # Round distances to a target precision to avoid the situation where
+    # if the distance is exactly 50ms (and strict=False) it erroneously
+    # doesn't match the notes because of precision issues.
+    onset_distances = np.around(onset_distances, decimals=4)
+    onset_hit_matrix = cmp_func(onset_distances, onset_tolerance)
+
+    # check for pitch matches
+    pitch_distances = np.abs(1200*np.subtract.outer(np.log2(ref_pitches),
+                                                    np.log2(est_pitches)))
+    pitch_hit_matrix = cmp_func(pitch_distances, pitch_tolerance)
+
+    # check for offset matches if offset_ratio is not None
+    if offset_ratio is not None:
+        offset_distances = np.abs(np.subtract.outer(ref_intervals[:, 1],
+                                                    est_intervals[:, 1]))
+        # Round distances to a target precision to avoid the situation where
+        # if the distance is exactly 50ms (and strict=False) it erroneously
+        # doesn't match the notes because of precision issues.
+        offset_distances = np.around(offset_distances, decimals=4)
```

Comment:
True, but it's generally bad form to have seemingly arbitrary constants scattered throughout a function.  Alternately, it could be moved up to a global constant; there's precedent for that in `beat`, `separation` and `chord`.